### PR TITLE
Remove Hashcons.Hobj

### PR DESCRIPTION
**Pull request by whitequark**
No functional change. See #7839 for motivation.

### CODE REVIEW

File: doc/sphinx/addendum/generalized-rewriting.rst [hunk 1]
```diff
@@ -106,7 +106,7 @@ argument.
 
 Morphisms can also be contravariant in one or more of their arguments.
 A morphism is contravariant on an argument associated to the relation
-instance :math`R` if it is covariant on the same argument when the inverse
+instance :math:`R` if it is covariant on the same argument when the inverse
```
Comment (SkySkimmer) [inline doc/sphinx/addendum/generalized-rewriting.rst]: unrelated change

**Review by ppedrot — APPROVED** [review]
If that's really dead code, doesn't matter.
